sql/core.go

```diff
@@ -283,8 +285,57 @@ type RowUpdater interface {
 // Database represents the database.
 type Database interface {
 	Nameable
-	// Tables returns the information of all tables.
-	Tables() map[string]Table
+
+	// GetTableInsensitive retrieves a table by it's name where capitalization does not matter
+	GetTableInsensitive(ctx context.Context, tblName string) (Table, bool, error)
+
+	// GetTableNames returns the table names of every table in the database
+	GetTableNames(ctx context.Context) ([]string, error)
+}
+
+// GetTableIncensitive implements a case insensitive map lookup for tables keyed off of the table name
+func GetTableInsensitive(tblName string, tables map[string]Table) (Table, bool) {
```

Comment:
This should look for an exact match before looking for a case insensitive match.

Reply:
It should also throw an error if there is more than one case-insensitive match.

Reply:
updated.  added a comment that if there is more than one case insensitive match that the one that is returned is not specified.